database.go

```diff
@@ -53,24 +54,48 @@ func (r *Recurser) ConvertToMap() map[string]interface{} {
 	}
 }
 
-func MapToStruct(m map[string]interface{}) Recurser {
+func MapToStruct(m map[string]interface{}) (Recurser, error) {
+	var id int64
+	switch v := m["id"].(type) {
+	case int:
+	case int8:
+	case int16:
+	case int32:
+	case int64:
+	case uint:
+	case uint8:
+	case uint16:
+	case uint32:
+	case uint64:
+		id = int64(v)
```

Comment:
[nit] I think the `uint64` case needs a bounds check because the `uint64` max value doesn't fit in an `int64` ([playground](https://go.dev/play/p/8J38hei8Kjg) just in case). I'd easily believe that the API's ID field is an auto-incrementing integer that's nowhere near this value (or likely to ever hit it), but it's the kind of unchecked cast that makes me nervous.

Reply:
Fair point -- sending a PR to update.

The RC API reports this as a raw integer, not a stringified int, so in theory it can be at most 2**53 - 1; still, that is more than 2**31-1.